core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -375,12 +375,9 @@ public void configureBaseApplication(Configurable<?> config) {
    * client.
    */
   public void configureBaseApplication(Configurable<?> config, Map<String, String> metricTags) {
-    RestConfig restConfig = getConfiguration();
+    registerJsonProvider(config, true);
```

Comment:
Yeah, that's what I had in mind. The exception mapper stuff split up a bit weirdly since the one for the `JsonProvider` is in `registerJsonProvider` instead of `registerExceptionMappers`, but otherwise breaking things down that way seems like a sane way to allow overriding the behavior without having to override everything.